oshi-core/src/main/java/oshi/software/os/unix/aix/AixOSProcess.java

```diff
@@ -290,24 +289,16 @@ public long getAffinityMask() {
 
     @Override
     public List<OSThread> getThreadDetails() {
-        List<OSThread> threads = new ArrayList<>();
-
         // Get process files in proc
         File directory = new File(String.format("/proc/%d/lwp", getProcessID()));
         File[] numericFiles = directory.listFiles(file -> Constants.DIGITS.matcher(file.getName()).matches());
         if (numericFiles == null) {
-            return threads;
+            return Collections.emptyList();
         }
 
-        // Iterate files
-        for (File lwpidFile : numericFiles) {
-            int lwpidNum = ParseUtil.parseIntOrDefault(lwpidFile.getName(), 0);
-            OSThread thread = new AixOSThread(getProcessID(), lwpidNum);
-            if (thread.getState() != INVALID) {
-                threads.add(thread);
-            }
-        }
-        return threads;
+        return Arrays.stream(numericFiles).parallel()
+            .map(lwpidFile -> new AixOSThread(getProcessID(), ParseUtil.parseIntOrDefault(lwpidFile.getName(), 0)))
+            .filter(thread -> thread.getState() != INVALID).collect(Collectors.toList());
```

Comment:
Nice job with the stream.  

We're re-using this invalid thread state check in multiple classes, so it would make sense to make it a constant on either the AbstractOSProcess parent or make it part of the API on OSProcess, similar to `OperatingSystem.VALID_PROCESS`.   Probably don't need a whole ThreadSorting and ThreadFiltering set of classes, though. WDYT?

Reply:
sounds good! I copied the format of how we have in `OperatingSystem` placed the constants in interface `OSThread` so it would kinda look the same, (but that way, seems like we have to keep the classes in there just like `OperatingSystem` tho...)